### ISSUE TITLE
EBM parameter search with Ray Tune

### ISSUE THREAD

**Issue by meriliis** [microsoft/interpret#55]
Hi!

I'm using Tune to search for optimal set of hyperparameters for different models. It works without issues for CatBoost and Keras models, however, I haven't been able to successfully run it with EBM on more than one CPU, let alone on GPU (it seems that EBM can't be run on GPU at all). When I haven't explicitly set `n_jobs=1`, I get this warning:

`UserWarning: Loky-backed parallel loops cannot be nested below threads, setting n_jobs=1`

Do you know is there something I'm missing or is it not possible to run EBM with Tune on multiple CPUs?

**Code to reproduce**
```
import ray
from ray import tune

ray.init(ignore_reinit_error=True)

import numpy as np
from interpret.glassbox import ExplainableBoostingClassifier

def train_ebm(config, reporter):
    n_samples = 100000
    x_train = np.random.rand(n_samples, 3)  # Random training data with 3 features
    y_train = np.random.randint(0, 2, n_samples)  # Random binary labels
    
    model = ExplainableBoostingClassifier(
        learning_rate=config['lr'], 
        n_jobs=12
    )
    
    model.fit(x_train, y_train)
    
    reporter()

ebm_experiment = tune.Experiment(
    name='ebm_test', 
    run=train_ebm, 
    num_samples=1,
    resources_per_trial={
        'gpu': 0,
        'cpu': 12
    },
    config={
        'lr': 0.01,
    }
)

trials = tune.run_experiments(ebm_experiment)
```

**Versions**
ray: 0.7.3
interpret: 0.1.15

**Comment by interpret-ml** [comment]
Hi Meri, 

This seems to be a limitation of our use of Joblib. When setting ```n_jobs``` to something other than 1, each EBM will parallelize across multiple cores on your machine. Ray also utilizes multiple cores to run different versions of models, and this nested parallelism isn't something the scheduler in joblib can handle for the type of backend we use. 

You might find it to be just as fast to tune by setting each EBM's ```njobs``` parameter to -1, and then running through your experiments sequentially. That way each individual EBM should fully utilize all the CPUs on your machine, so there aren't "wasted" cores as you are running your experiments.

Right now in this initial release, we don't have a GPU option for training EBMs, but it is something we're investigating for future releases. Thanks for expressing your interest, it's good to know it would be useful!

-InterpretML Team

**Comment by tmontana** [comment]
Optuna works fine for this use case FYI

https://optuna.org/
